tests/golos.publication_rewards_tests.cpp

```diff
@@ -1287,4 +1287,73 @@ BOOST_FIXTURE_TEST_CASE(golos_delegators_test, reward_calcs_tester) try {
     check();
 } FC_LOG_AND_RETHROW()
 
+BOOST_FIXTURE_TEST_CASE(posting_bw_penalty, reward_calcs_tester) try {
+    BOOST_TEST_MESSAGE("Posting bw penalty testing");
+    int64_t maxfp = std::numeric_limits<fixp_t>::max();
+    auto bignum = 500000000000;
+    auto five_min = 5*60;
+    auto window = 5000;
+    auto pb_cutoff = cfg::_100percent*4;
+    auto reward_weight_delta = 0.0001;
+    init(bignum, 500000);
+    produce_blocks();
+    
+    name action = "calcrwrdwt"_n;
+    auto auth = authority(1, {}, {
+        {.permission = {golos::config::charge_name, config::eosio_code_name}, .weight = 1}
+    });
+    set_authority(_code, action, auth, "active");
+    link_authority(_code, _code, action, action);
```

Comment:
It's better to make in https://github.com/GolosChain/golos.contracts/blob/29969f48ff43c07e701736a920e7ac36542c9d6b/tests/golos.posting_test_api.hpp#L16-L24 (pass `charge_name` as parameter)